### ISSUE TITLE
Have the different settings from "Settings" dialogue logged.

### ISSUE THREAD

**Issue by GrantSP** [LePresidente/modorganizer#150]
In diagnosing an [issue with MO 1.3.11 and LOOT that was first described in the LOOT repo](https://github.com/loot/loot/issues/855), it was finally found to be an esoteric setting in the "Workarounds" tab that was the culprit. The actual setting was: **Hide inactive plugins**.
As I had never seen anyone use that setting I didn't ask to see if it was enabled.

If the settings could be logged in the mo_interface.log,  or elsewhere, this sort of thing will be easier to track down and resolve.

**Comment by erasmux** [comment]
Should be fixed in the 26/12 dev build on the discord channel. Please test.

**Comment by GrantSP** [comment]
Excellent. This is a sample of what I see.

> 17:05:26 [D] Loaded settings:
> 17:05:26 [D]   compact_downloads=true
> 17:05:26 [D]   crash_dumps_max=5
> 17:05:26 [D]   crash_dumps_type=1
> 17:05:26 [D]   display_foreign=true
> 17:05:26 [D]   download_directory=<PathToGameDownloads>
> 17:05:26 [D]   force_enable_core_files=true
> 17:05:26 [D]   hide_unchecked_plugins=true
> 17:05:26 [D]   language=
> 17:05:26 [D]   load_mechanism=0
> 17:05:26 [D]   log_level=0
> 17:05:26 [D]   meta_downloads=true
> 17:05:26 [D]   nexus_login=true
> 17:05:26 [D]   nexus_password=*************
> 17:05:26 [D]   nexus_username=************
> 17:05:26 [D]   nmm_version=0.61.13
> 17:05:26 [D]   offline_mode=false
> 17:05:26 [D]   style=dracula.qss
> 17:05:26 [D]   use_prereleases=false
> 17:05:26 [D]   use_proxy=false

I wonder if it is possible to order them as they appear in the settings dialogue?
Maybe with sub-headings? Not a big deal but if doable it would be fantastic.
ie.

> 17:05:26 [D] Loaded settings:
> 17:05:26 [D] General:
> 17:05:26 [D]   language=
> 17:05:26 [D]   style=dracula.qss
> 17:05:26 [D]   use_prereleases=false
> 17:05:26 [D]   compact_downloads=true
> 17:05:26 [D]   meta_downloads=true
> 17:05:26 [D] Paths:
> 17:05:26 [D]   download_directory=<PathToGameDownloads>
> 17:05:26 [D] Nexus:
> 17:05:26 [D]   nexus_login=true
> 17:05:26 [D]   nexus_password=***************
> 17:05:26 [D]   nexus_username=**************
> 17:05:26 [D]   offline_mode=false
> 17:05:26 [D]   use_proxy=false
> 17:05:26 [D] Workarounds:
> 17:05:26 [D]   load_mechanism=0
> 17:05:26 [D]   nmm_version=0.61.13
> 17:05:26 [D]   hide_unchecked_plugins=true
> 17:05:26 [D]   force_enable_core_files=true
> 17:05:26 [D]   display_foreign=true
> 17:05:26 [D] Diagnostics:
> 17:05:26 [D]   log_level=0
> 17:05:26 [D]   crash_dumps_type=1
> 17:05:26 [D]   crash_dumps_max=5
> 

I didn't see anything for the "Steam" tab. I didn't have anything in that anyway so I don't know if that is why. Not really needed for diagnostic reasons though so no need to add it in.
I think what you've already done with this will go a long way.
Thank you for putting in the time. Cheers.
What ever you decide to do I truly appreciate it.

**Comment by erasmux** [comment]
Yeah I basically put the minimal effort into this as this is just for diagnostics. This just dumps the [Settings] sections from the modorganizer.ini. If you set the user/pass for steam it should output that with the 26/12 build (although I didn't test).

One side note: after the 26/12 build I noticed that the nexus_username/password are logged with everything else. For privacy reasons, that info has been removed from the log for future builds (the pull request already includes this change). This change should also remove the steam user/pass and any other future keys with "username" or "password" in their name.
